src/com/tictacgo/data/Piece.java

```diff
@@ -206,7 +205,7 @@ public Player getPlayer() {
 
     public void setPlayer(Player player) {
         this.player = player;
-        updateImageResource();
+        updateImageResource(true);
```

Comment:
Yes, I originally branched before those changes were made I think. Those should get changed automatically on merge.